parse/parser_internal_test.go

```diff
@@ -1,7 +1,86 @@
+/*
+ * Copyright 2018 De-labtory
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
 package parse
 
 import "testing"
 
+type MockTokenBuffer struct {
+	buf []Token
+	sp  int
+}
+
+func (m *MockTokenBuffer) Read() Token {
+	ret := m.buf[m.sp]
+	m.sp++
+	return ret
+}
+
+func (m *MockTokenBuffer) Peek(n peekNumber) Token {
+	return m.buf[n]
+}
+
+func TestParse_curTokenIs(t *testing.T) {
+	test := MockTokenBuffer{
+		buf: []Token{
```

Comment:
It would be better to name the variable tokenBuf not test.
`tokenBuf := `

Reply:
Also modified.